src/dali_executor/io_descriptor.h

```diff
@@ -56,11 +56,29 @@ template<typename T>
 struct IODescr {
   IOMeta meta{};
   std::vector<BufferDescr<T>> buffers{};
+
+  void append(const IODescr &other) {
+    if (meta.shape.num_samples() == 0) {
+      meta = other.meta;
```

Comment:
Could you add documentation to this function?

Reply:
added